### PR TITLE
Feature extract method Update 2

### PR DESCRIPTION
**Pull request by bclothier**
Continuation of previous PR that was accidentally merged prematurely.

### CODE REVIEW

File: RetailCoder.VBE/UI/RubberduckUI.resx [hunk 1]
```diff
@@ -2176,4 +2176,7 @@ Would you like to import them to Rubberduck?</value>
   <data name="ExtractMethod_NoneSelected" xml:space="preserve">
     <value>(None)</value>
   </data>
-</root>
\ No newline at end of file
+  <data name="ExtractMethod_CompilationDirectivesWarning" xml:space="preserve">
+    <value>The module contains compilation directives (e.g. #Const, #If, ##lse or #End If). The refactoring can only operate on the active blocks. You must manually update the inactive blocks after the refactoring.</value>
```
Comment (Vogel612) [inline RetailCoder.VBE/UI/RubberduckUI.resx]: typo ##lse should be #Else

File: Rubberduck.VBEEditor/Extensions/CodeModuleExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            return regex.Matches(rawCode).Count > 0;
+        }
+
+        public static bool ContainsCompilationDirectives(this ICodeModule codeModule)
+        {
+            var rawCode = string.Join(Environment.NewLine,
+                codeModule.GetLines(1, codeModule.CountOfLines));
+            var regex = new Regex(@"^(\s?)+#", RegexOptions.Multiline);
+            return (regex.Matches(rawCode).Count > 0);
+        }
```
Comment (Vogel612) [inline Rubberduck.VBEEditor/Extensions/CodeModuleExtensions.cs]: I think I'd rewrite this to call something like:

```
private static bool ContainsCompilationDirectives(string rawCode) 
{
    // local static may not be supported, if it isn't extract it into a constant
    static var regex = new Regex(@"^\s*#", RegexOptions.Multiline);
    return regex.Matches(rawCode).Count > 0;
}
```

and just call this from the other extension methods

File: Rubberduck.VBEEditor/Extensions/CodeModuleExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        public static bool ContainsCompilationDirectives(this ICodeModule codeModule)
+        {
+            var rawCode = string.Join(Environment.NewLine,
+                codeModule.GetLines(1, codeModule.CountOfLines));
+            return ContainsCompilationDirectives(rawCode);
+        }
+
+        private static bool ContainsCompilationDirectives(string rawCode)
+        {
+            const string regexExpression = @"^(\s?)+#";
```
Comment (Vogel612) [inline Rubberduck.VBEEditor/Extensions/CodeModuleExtensions.cs]: I did change the regex. Why did you not?

Comment (bclothier) [reply Rubberduck.VBEEditor/Extensions/CodeModuleExtensions.cs]: Apologies - it's only because I overlooked it was changed. My bad.

**Pull request merged**